### PR TITLE
Add base classes and fragment identifiers for vocabulary.

### PR DESCRIPTION
**Pull request by msporny**
This PR attempts to address part of issue #1263 by defining base classes (and their anchors) in the VCDM specification.

***

### :boom: Error: 500 Internal Server Error :boom: ###

[PR Preview](https://github.com/tobie/pr-preview#pr-preview) failed to build. _(Last tried on Sep 26, 2023, 8:39 PM UTC)_.

<details>
<summary>More</summary>

PR Preview relies on a number of web services to run. There seems to be an issue with the following one:

:rotating_light: [Spec Generator](https://www.w3.org/2015/labs/) - Spec Generator is the web service used to build specs that rely on ReSpec.

:link: [Related URL](https://labs.w3.org/spec-generator/?type=respec&url=https%3A%2F%2Fraw.githubusercontent.com%2Fw3c%2Fvc-data-model%2Ff5b8ac9e937acf27a8e63bcef30bfb8d0cc1f434%2Findex.html%3FisPreview%3Dtrue)

```

😭  Sorry, there was an error generating the HTML. Please report this issue!
Specification: http://labs.w3.org/spec-generator/uploads/teRudZ/index.html?isPreview=true%3FisPreview%3Dtrue&publishDate=2023-09-26
ReSpec version: 34.1.8
File a bug: https://github.com/w3c/respec/
Error: undefined

```

_If you don't have enough information above to solve the error by yourself (or to understand to which web service the error is related to, if any), please [file an issue](https://github.com/tobie/pr-preview/issues/new?title=Error%20not%20surfaced%20properly&body=See%20w3c/vc-data-model%231272.)._
</details>

### CODE REVIEW

**Review by iherman — COMMENTED** [review]
Would it be possible to, in the same PR, make the changes on the vocabulary.yml file, too? For each of those classes a `defined_by` statement should replace the comment field.... This would save us yet another PR/merge round.

(I can make a PR against this one if you prefer.)

File: index.html [hunk 1]
```diff
@@ -5963,6 +5963,104 @@ <h3>Vocabularies</h3>
...
+            <tr id="bc-credential-evidence">
+              <td>
+`CredentialEvidence`
+              </td>
+              <td>
+Serves as a supercass for specific evidence types that are placed into the
+<a href="#evidence">evidence</a> property. <span class="issue atrisk">This
+superclass is at risk and will be removed if at least two independent
+implementations for the super class are not identified by the end of the
+Candidate Recommendation phase.</span>
```
Comment (TallTed) [inline index.html:5988]:
```suggestion
Serves as a superclass for specific evidence types that are placed into the
<a href="#evidence">evidence</a> property. <span class="issue atrisk">This
superclass is at risk and will be removed if at least two independent
implementations for the superclass are not identified by the end of the
Candidate Recommendation phase.</span>
```

File: index.html [hunk 2]
```diff
@@ -5963,6 +5963,104 @@ <h3>Vocabularies</h3>
 cryptographic hash.
         </p>
 
+        <p>
+The following base classes are defined in this specification for processors
+and other specifications that benefit from the definition of base classes:
```
Comment (TallTed) [inline index.html:5968]:
```suggestion
and other specifications that benefit from such definitions:
```

File: index.html [hunk 3]
```diff
@@ -5963,6 +5963,104 @@ <h3>Vocabularies</h3>
...
+implementations for the super class are not identified by the end of the
+Candidate Recommendation phase.</span>
+              </td>
+            </tr>
+            <tr id="bc-credential-schema">
+              <td>
+`CredentialSchema`
+              </td>
+              <td>
+Serves as a supercass for specific schema types that are placed into the
```
Comment (TallTed) [inline index.html:5996]:
```suggestion
Serves as a superclass for specific schema types that are placed into the
```

File: index.html [hunk 4]
```diff
@@ -5963,6 +5963,104 @@ <h3>Vocabularies</h3>
...
+<a href="#data-schemas">credentialSchema</a> property.
+              </td>
+            </tr>
+            <tr id="bc-credential-status">
+              <td>
+`CredentialStatus`
+              </td>
+              <td>
+Serves as a supercass for specific credential status types that are placed into
```
Comment (TallTed) [inline index.html:6005]:
```suggestion
Serves as a superclass for specific credential status types that are placed into
```

File: index.html [hunk 5]
```diff
@@ -5963,6 +5963,104 @@ <h3>Vocabularies</h3>
...
+            <tr id="bc-confidence-method">
+              <td>
+`ConfidenceMethod`
+              </td>
+              <td>
+Serves as a supercass for specific confidence method types that are placed into
+the `confidenceMethod` property.
+<span class="issue atrisk">This superclass is at risk and will be removed if
+at least two independent implementations for the super class are not identified
+by the end of the Candidate Recommendation phase.
```
Comment (TallTed) [inline index.html:6018]:
```suggestion
Serves as a superclass for specific confidence method types that are placed into
the `confidenceMethod` property.
<span class="issue atrisk">This superclass is at risk and will be removed if
at least two independent implementations for the superclass are not identified
by the end of the Candidate Recommendation phase.
```

File: index.html [hunk 6]
```diff
@@ -5963,6 +5963,104 @@ <h3>Vocabularies</h3>
...
+              <td>
+`RefreshService`
+              </td>
+              <td>
+Serves as a supercass for specific refresh service types that are placed into
+the <a href="#refreshing">credentialRefresh</a> property.
+<span class="issue atrisk">This superclass is at risk and will be removed if
+at least two independent implementations for the super class are not identified
+by the end of the Candidate Recommendation phase.
+</span>
```
Comment (TallTed) [inline index.html:6032]:
```suggestion
Serves as a superclass for specific refresh service types that are placed into
the <a href="#refreshing">credentialRefresh</a> property.
<span class="issue atrisk">This superclass is at risk and will be removed if
at least two independent implementations for the superclass are not identified
by the end of the Candidate Recommendation phase.
</span>
```

File: index.html [hunk 7]
```diff
@@ -5963,6 +5963,104 @@ <h3>Vocabularies</h3>
...
+            <tr id="bc-render-method">
+              <td>
+`RenderMethod`
+              </td>
+              <td>
+Serves as a supercass for specific render method types that are placed into
+the `renderMethod` property.
+<span class="issue atrisk">This superclass is at risk and will be removed if
+at least two independent implementations for the super class are not identified
+by the end of the Candidate Recommendation phase.
```
Comment (TallTed) [inline index.html:6044]:
```suggestion
Serves as a superclass for specific render method types that are placed into
the `renderMethod` property.
<span class="issue atrisk">This superclass is at risk and will be removed if
at least two independent implementations for the superclass are not identified
by the end of the Candidate Recommendation phase.
```

File: index.html [hunk 8]
```diff
@@ -5963,6 +5963,104 @@ <h3>Vocabularies</h3>
...
+            <tr id="bc-terms-of-use">
+              <td>
+`TermsOfUse`
+              </td>
+              <td>
+Serves as a supercass for specific terms of use types that are placed into
+the <a href="#terms-of-use">termsOfUse</a> property.
+<span class="issue atrisk">This superclass is at risk and will be removed if
+at least two independent implementations for the super class are not identified
+by the end of the Candidate Recommendation phase.
```
Comment (TallTed) [inline index.html:6057]:
```suggestion
Serves as a superclass for specific terms of use types that are placed into
the <a href="#terms-of-use">termsOfUse</a> property.
<span class="issue atrisk">This superclass is at risk and will be removed if
at least two independent implementations for the superclass are not identified
by the end of the Candidate Recommendation phase.
```

**Review by TallTed — CHANGES_REQUESTED** [review]

**Review by OR13 — APPROVED** [review]

**Review by dlongley — APPROVED** [review]

**Comment by iherman** [comment]
> Would it be possible to, in the same PR, make the changes on the vocabulary.yml file, too? For each of those classes a `defined_by` statement should replace the comment field.... This would save us yet another PR/merge round.
> 
> (I can make a PR against this one if you prefer.)

Putting my money where my mouth is... I have created #1278 to do this.

@msporny, if you are o.k. with this, the best would be to merge #1278 right away into this one, so that all the possible comments would be at one place. I just did not want to directly edit your PR.

**Review by decentralgabe — APPROVED** [review]

**Review by David-Chadwick — APPROVED** [review]

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-09-14-vcwg#section1-6) on 2023-09-14

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.6. Add base classes and fragment identifiers for vocabulary. (pr vc-data-model#1272)

_See github pull request [vc-data-model#1272](https://github.com/w3c/vc-data-model/pull/1272)._




**Brent Zundel:** primarily editorial, raised a few days ago.  

**Manu Sporny:** Yes to all of that, just need to merge TallTed's changes.  

_See github pull request [vc-data-model#1278](https://github.com/w3c/vc-data-model/pull/1278)._




**Ivan Herman:** For the record, I did 1278 yesterday.  
� it's on top of 1272.  

**Manu Sporny:** agree to merge on top of it.  

**Brent Zundel:** merged, back to 1272.  
� still have 3 more days for reviews.  
� manu to look at suggestions made, anticipating that the PR will be merged soon.  

> *Sebastian Crane:* What I was saying is that the ease of merging Ivan's PR illustrates how useful it is to push to a 'personal' branch within the W3C repository, not to your GitHub fork. It is not possible to push to someone else's GitHub fork, so that discussion would have involved lots of git commands if Manu hadn't pushed to the main repository.

</details>

**Comment by msporny** [comment]
Normative, multiple reviews, changes requested and made, no objections, merging.

**Pull request merged**